pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1307,15 +1307,26 @@ func (r *ReconcileInfinispan) gracefulShutdownReq(ispn *infinispanv1.Infinispan,
 	logger.Info("Sending graceful shutdown request")
 	// send a graceful shutdown to the first ready pod
 	// if there's no ready pod we're in trouble
+
 	for _, pod := range podList.Items {
 		if kube.IsPodReady(pod) {
 			err := cluster.GracefulShutdown(pod.GetName())
 			if err != nil {
-				logger.Error(err, "failed to exec shutdown command on pod")
-				continue
+				logger.Error(err, "Error encountered on cluster shutdown")
+
+				// ISPN-13141 causes GracefulShutdown to fail if there are issues with the server
+				logger.Info("Cluster Shutdown failed. Attempting to execute GracefulShutdownTask")
+				if err := cluster.GracefulShutdownTask(pod.GetName()); err != nil {
+					logger.Error(err, "Error encountered on GracefulShutdownTask")
+				}
```

Comment:
if `GracefulShutdownTask` fails and then all the pods are deleted, the js is not successfully executed on the cluster. Is this ok?

Reply:
Hmm good point. I will update this so that it tries executing the task on the next ready pod. If the task fails on all pods, then I think we have to proceed with deleting the pods and updating the condition so that the upgrade can continue.

Previously if `GracefulShutdown` failed, we return `nil` which resulted in a strange state where the wrong conditions aren't in place and the upgrade can't proceed or recover.